src/category_theory/abelian/exact.lean

```diff
@@ -304,6 +309,191 @@ instance reflects_exact_sequences_of_preserves_zero_morphisms_of_faithful (F : C
     rw [F.map_comp, ← hk, ← hl, category.assoc, reassoc_of hfg.2, zero_comp, comp_zero]
   end }
 
+end
+
+section
+variables {D : Type u₂} [category.{v₁} D] [abelian D]  -- TODO: Generalize universe level
+variables (F : C ⥤ D) [preserves_zero_morphisms F]
+
+/--
+A functor from a preadditive category to an abelian category which preserves kernels
+preserves finite products.
+-/
+def is_limit_map_cone_binary_fan_of_preserves_kernels {X Y Z : C} (π₁ : Z ⟶ X) (π₂ : Z ⟶ Y)
+  [preserves_limit (parallel_pair π₂ 0) F] (i : is_limit (binary_fan.mk π₁ π₂)) :
+  is_limit (F.map_cone (binary_fan.mk π₁ π₂)) :=
+let bc := binary_bicone.of_limit_cone i in
+let ibc := bicone_is_bilimit_of_limit_cone_of_is_limit i in
+begin
+  haveI : preserves_limit (parallel_pair bc.snd 0) F, { simpa },
+  let i_f : is_limit bc.snd_kernel_fork := binary_bicone.is_limit_snd_kernel_fork i,
+  have hex : exact (F.map bc.inl) (F.map bc.snd),
+  { fapply abelian.exact_of_is_kernel,
+    { rw [←F.map_comp, bc.inl_snd, F.map_zero] },
+    { exact (is_limit_map_cone_fork_equiv' F _).to_fun (is_limit_of_preserves F i_f) } },
+  exact (is_limit_map_cone_binary_fan_equiv F π₁ π₂).inv_fun
+    (is_bilimit_binary_bicone_of_split_mono_of_cokernel
+      (abelian.is_colimit_of_exact_of_epi (F.map bc.inl) (F.map bc.snd) hex)).is_limit
+end
+
+instance preserves_binary_product_of_preserves_kernels
+  [pres_kernels : ∀ {X Y} (f : X ⟶ Y), preserves_limit (parallel_pair f 0) F] {X Y : C} :
+  preserves_limit (pair X Y) F :=
+{ preserves := λ c hc, is_limit.of_iso_limit
+    (is_limit_map_cone_binary_fan_of_preserves_kernels F _ _
+      (is_limit.of_iso_limit hc (iso_binary_fan_mk c)))
+    ((cones.functoriality _ F).map_iso (iso_binary_fan_mk c).symm) }
+
+instance preserves_binary_products_of_preserves_kernels
+  [pres_kernels : ∀ {X Y} (f : X ⟶ Y), preserves_limit (parallel_pair f 0) F] :
+  preserves_limits_of_shape (discrete walking_pair) F :=
+{ preserves_limit := λ p, preserves_limit_of_iso_diagram F (diagram_iso_pair p).symm }
+
+/--
+A functor from a preadditive category into an abelian category preserves the equalizer of two
+morphisms if it preserves all kernels. -/
+def preserves_equalizer_of_preserves_kernels
+  [pres_kernels : ∀ {X Y} (f : X ⟶ Y), preserves_limit (parallel_pair f 0) F] {X Y : C}
```

Comment:
I don't think you should be naming instance arguments unnecessarily; it's just noise.

Reply:
Ah, thanks, that's just remains from an earlier state where manual reference was needed.